pylint/checkers/base/name_checker/checker.py

```diff
@@ -545,42 +560,60 @@ def _assigns_typevar(node: Optional[nodes.NodeNG]) -> bool:
                 return True
         return False
 
-    def _check_typevar_variance(self, name: str, node: nodes.AssignName) -> None:
-        """Check if a TypeVar has a variance and if it's included in the name.
-
-        Returns the args for the message to be displayed.
-        """
+    def _check_typevar(self, name: str, node: nodes.AssignName) -> None:
+        """Check for TypeVar lint violations."""
         if isinstance(node.parent, nodes.Assign):
             keywords = node.assign_type().value.keywords
         elif isinstance(node.parent, nodes.Tuple):
             keywords = (
                 node.assign_type().value.elts[node.parent.elts.index(node)].keywords
             )
 
+        variance = TypeVarVariance.invariant
         for kw in keywords:
             if kw.arg == "covariant" and kw.value.value:
-                if not name.endswith("_co"):
-                    suggest_name = f"{re.sub('_contra$', '', name)}_co"
-                    self.add_message(
-                        "typevar-name-incorrect-variance",
-                        node=node,
-                        args=(name, "covariant", suggest_name),
-                        confidence=interfaces.INFERENCE,
-                    )
-                return
-
-            if kw.arg == "contravariant" and kw.value.value:
-                if not name.endswith("_contra"):
-                    suggest_name = f"{re.sub('_co$', '', name)}_contra"
-                    self.add_message(
-                        "typevar-name-incorrect-variance",
-                        node=node,
-                        args=(name, "contravariant", suggest_name),
-                        confidence=interfaces.INFERENCE,
+                variance = (
+                    TypeVarVariance.covariant
+                    if variance
+                    not in (
```

Comment:
This feels so nitpicky, but in the GitHub `diff` viewer the current code is a little unreadable imo:
Can we move the check for `TypeVarVariance.double_variant` to the if that checks the `kw.arg`? I think we can do it as the first check and skip the complete if statement if we already encounter a double_variant.